hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaDeploymentManagement.java

```diff
@@ -881,4 +894,117 @@ private static RetryTemplate createRetryTemplate() {
 
         return template;
     }
+
+    @Override
+    // No transaction, will be created per handled distribution set
+    @Transactional(propagation = Propagation.NEVER)
+    public void invalidateDistributionSet(final DistributionSetInvalidation distributionSetInvalidation) {
+        LOG.debug("Invalidate distribution sets {}", distributionSetInvalidation.getDistributionSetIds());
+        final String tenant = tenantAware.getCurrentTenant();
+
+        if (shouldRolloutsBeCanceled(distributionSetInvalidation.getCancelationType(),
+                distributionSetInvalidation.isCancelRollouts())) {
+            final String handlerId = JpaRolloutManagement.createRolloutLockKey(tenant);
+            final Lock lock = lockRegistry.obtain(handlerId);
+            try {
+                if (!lock.tryLock(repositoryProperties.getDsInvalidationLockTimeout(), TimeUnit.SECONDS)) {
+                    throw new StopRolloutException("Timeout while trying to invalidate distribution sets");
+                }
+                try {
+                    invalidateDistributionSetsInTransaction(distributionSetInvalidation, tenant);
+                } finally {
+                    lock.unlock();
+                }
+            } catch (final InterruptedException e) {
+                LOG.error("InterruptedException while invalidating distribution sets {}!",
+                        distributionSetInvalidation.getDistributionSetIds(), e);
+                Thread.currentThread().interrupt();
+            }
+        } else {
+            // no lock is needed as no rollout will be stopped
+            invalidateDistributionSetsInTransaction(distributionSetInvalidation, tenant);
+        }
+    }
+
+    private void invalidateDistributionSetsInTransaction(final DistributionSetInvalidation distributionSetInvalidation,
+            final String tenant) {
+        DeploymentHelper.runInNewTransaction(txManager, tenant + "-invalidateDS", status -> {
+            distributionSetInvalidation.getDistributionSetIds().forEach(setId -> invalidateDistributionSet(setId,
+                    distributionSetInvalidation.getCancelationType(), distributionSetInvalidation.isCancelRollouts()));
+            return 0;
+        });
+    }
+
+    private void invalidateDistributionSet(final long setId, final CancelationType cancelationType,
+            final boolean cancelRollouts) {
+        final JpaDistributionSet set = (JpaDistributionSet) distributionSetManagement.getValidAndComplete(setId);
```

Comment:
consider defining own "invalidateDs" method under DistributionSetManagement instead of calling 
`distributionSetRepository.save` here